### PR TITLE
Builder handles empty job on component

### PR DESCRIPTION
**Pull request by kleesc**
**Issue:** https://issues.redhat.com/browse/PROJQUAY-974

**Changelog:** 

**Docs:** 

**Testing:** 

**Details:** 

------
(_This section may be deleted._)
**All fields are required.** If a field is not applicable (eg. no relevant CHANGELOG.md), specify "none" or "n/a".

Issue: This is the PROJQUAY jira reference. Pull-request title must start with issue name "PROJQUAY-1234 - ".

Changelog: One line description to be added to CHANGELOG.md during release builds. Typically starts with "Added:", "Fixed:", "Note:", etc.

Docs: Detailed description of changes necessary to docs.projectquay.io. Examples would be addition of config.yaml, indication of UI changes and screenshot impact, and changes in behavior of features.

Testing: Detailed description of how to test changes manually. This section combined with the _Docs_ section above must be sufficiently clear for full test cases to be performed.

Details: Other information meant for pull-request reviewers and developers.

### CODE REVIEW

File: buildman/component/buildcomponent.py [hunk 1]
```diff
@@ -93,6 +93,10 @@ def start_build(self, build_job):
         """
         Starts a build.
         """
+        if build_job is None:
+            logger.error("Error starting build on realm %s, no job given", self,builder_realm)
```
Comment (josephschorr) [inline buildman/component/buildcomponent.py:97]: `self,` -> `self.`

File: buildman/manager/ephemeral.py [hunk 2]
```diff
@@ -497,8 +497,8 @@ def schedule(self, build_job):
             # Check if we can use this executor based on its whitelist, by namespace.
             namespace = build_job.namespace
             if not executor.allowed_for_namespace(namespace):
-                logger.debug(
-                    "Job %s (namespace: %s) cannot use executor %s",
+                logger.error(
```
Comment (josephschorr) [inline buildman/manager/ephemeral.py:500]: This should be a `warning`

File: buildman/manager/ephemeral.py [hunk 3]
```diff
@@ -523,8 +523,12 @@ def schedule(self, build_job):
 
             try:
                 execution_id = yield From(executor.start_builder(realm, token, build_uuid))
-            except:
-                logger.exception("Exception when starting builder for job: %s", build_uuid)
+            except exception as e:
```
Comment (josephschorr) [inline buildman/manager/ephemeral.py:526]: Can you also add a `logger.error` into the actual Kub start exception handler?

File: buildman/component/buildcomponent.py [hunk 4]
```diff
@@ -668,6 +676,10 @@ def _timeout(self):
         yield From(self._set_status(ComponentStatus.TIMED_OUT))
         logger.warning("Build component with realm %s has timed out", self.builder_realm)
 
+        if self._current_job is None:
+            logger.error("Error trying to time out build. Build on realm %s has already been completed", self.builder_realm)
+            raise Return()
+
         build_id = self._current_job.repo_build.uuid
```
Comment (josephschorr) [inline buildman/component/buildcomponent.py:683]: Interesting that we check if `self._current_job` is not None a few lines later... looks like this build_id might need to be in that block

File: buildman/manager/ephemeral.py (hunk 3 above)
Comment (josephschorr) [inline buildman/manager/ephemeral.py:526]: Why lowercase `exception`?

Comment (kleesc) [reply buildman/manager/ephemeral.py]: Fixed

File: buildman/component/buildcomponent.py (hunk 4 above)
Comment (kleesc) [reply buildman/component/buildcomponent.py]: Moved the build_id it in the if block

File: buildman/manager/ephemeral.py (hunk 2 above)
Comment (kleesc) [reply buildman/manager/ephemeral.py]: Done

File: buildman/component/buildcomponent.py (hunk 1 above)
Comment (kleesc) [reply buildman/component/buildcomponent.py]: Fixed

File: buildman/manager/ephemeral.py (hunk 3 above)
Comment (kleesc) [reply buildman/manager/ephemeral.py]: Done

**Pull request merged**